lib/util/json-writer/editor-config.ts

```diff
@@ -0,0 +1,100 @@
+import { KnownProps, parseString } from 'editorconfig';
+import type { ParseStringResult } from 'editorconfig/src/lib/ini';
+import globToRegExp from 'glob-to-regexp';
+import { logger } from '../../logger/index';
+import { localPathExists, readLocalFile } from '../fs';
+import type { CodeFormat } from './code-format';
+import { IndentationType } from './indentation-type';
+
+export class EditorConfig {
+  private readonly formats: Map<RegExp, KnownProps> = new Map();
+
+  private static instance?: EditorConfig;
```

Comment:
Please use memory cache instead, so we can auto cleanup after repo processing

https://github.com/renovatebot/renovate/blob/095f2f8240ca90d3dc61016cb97b84d8e853c9b8/lib/util/cache/memory/index.ts#L15

Reply:
In one nice day we will receive collision of names with this approach